File: solutions/scala/bob/1/src/main/scala/Bob.scala
object Bob {

  def isQuestion(s: String): Boolean = s.endsWith("?")
  def isSilence(s: String): Boolean = s == ""
  def isYelling(s: String): Boolean = {
    val test = s.replaceAll("[^A-Za-z]", "")
    test.matches("[A-Z]+")
  }

  def response(statement: String): String = {
    statement.trim match {
      case s if isSilence(s) => "Fine. Be that way!"
      case s if isQuestion(s) && isYelling(s) => "Calm down, I know what I'm doing!"
      case s if isQuestion(s) => "Sure."
      case s if isYelling(s) => "Whoa, chill out!"
      case _ => "Whatever."
    }
  }

}

File: solutions/typescript/bob/1/bob.ts
export function hey(message: string): string {
  const trimmed = message.trim()

  const isEmpty = trimmed === ""
  const isQuestion = trimmed.endsWith("?")
  const isYelled =
    /\p{L}/u.test(trimmed) && trimmed === trimmed.toLocaleUpperCase()

  if (isEmpty) return "Fine. Be that way!"
  if (isYelled && isQuestion) return "Calm down, I know what I'm doing!"
  if (isYelled) return "Whoa, chill out!"
  if (isQuestion) return "Sure."
  return "Whatever."
}
